src/main/kotlin/lotto/model/Lotto.kt

```diff
@@ -1,16 +1,25 @@
 package lotto.model
 
-class Lotto(val numbers: List<Int>) {
+class Lotto(val numbers: Set<Int>) {
 
     init {
         require(numbers.size == NUMBER_COUNT) {
             "6개의 숫자를 선택해주세요."
         }
     }
 
-    fun convertPrize(winnerNumbers: Lotto): Prize {
+    fun convertPrize(winnerNumbers: Lotto, bonusBall: Int): Prize {
         val matchCount = numbers.count { winnerNumbers.numbers.contains(it) }
-        return Prize.findByMatchCount(matchCount)
+
+        return when {
+            matchCount != Prize.SECOND.matchCount -> {
+                Prize.findByMatchCount(matchCount = matchCount, isMatchedBonusNumber = false)
+            }
+            numbers.filterNot { winnerNumbers.numbers.contains(it) }.first() == bonusBall -> {
+                Prize.findByMatchCount(matchCount = matchCount, isMatchedBonusNumber = true)
+            }
+            else -> Prize.findByMatchCount(matchCount = matchCount, isMatchedBonusNumber = false)
+        }
```

Comment:
github 상으로만 코드를 파악해, 제가잘못 이해했을수도 있는데, 보너스 번호를 이용해 당첨자를 구하는 부분이 너무 복잡한 것 같습니다.

아래와 같이 한줄로 처리할 수 있지 않을까요?

```suggestion
        return Prize.findByMatchCount(matchCount = matchCount, isMatchedBonusNumber = this.numbers.contains(bonusBall))
```

추가로

```kotlin
numbers.filterNot { winnerNumbers.numbers.contains(it) }.first() == bonusBall
```

위 로직의 경우 winnerNumbers에 포함되지 않은 numbers의 값의 첫번째를 반환하는 것 같은데 첫번째가 아닌 두번째 값이 bonusBall이 될수도 있지 않을까요?

Reply:
구현 의도는 아래와 같습니다.

when 절의 첫 번째 항목에서 matchCount 가 5가 아닌 경우 먼저 검사합니다.
만약 해당 경우에 만족하지 못한다면 matchCount 가 5 인 경우입니다.
matchCount 가 5인 상황에서 `numbers.filterNot { winnerNumbers.numbers.contains(it) }` 을 하게 되면 size 는 무조건 1이 되고, 해서 first() 를 사용하여 보너스 볼과 비교하였습니다.

그런데 제가 구현한 로직을 한 줄로 정리가 가능하다니 대단하십니다!:+1:

수정하였습니다.
[2]

Reply:
아 그렇군요, 이 로직을 돌리는 시점이 이미 matchCount가 5인 경우기 때문에 first를 해도 되겠네요!

github 상으로만 보다 보니 해당 부분을 놓쳤습니다 ㅠㅠ

```
numbers.filterNot { winnerNumbers.numbers.contains(it) }.first() == bonusBall
```

피드백 반영해주신거 확인했습니다